src/build/app/settings.rs

```diff
@@ -655,6 +658,39 @@ pub enum AppSettings {
     /// This can be useful if there are many values, or they are explained elsewhere.
     HidePossibleValuesInHelp,
 
+    /// Tells `clap` to panic if help strings are omitted
+    ///  
+    /// # Examples
+    ///
+    /// ```rust
+    /// # use clap::{App, Arg, AppSettings};
+    /// App::new("myprog")
+    ///     .setting(AppSettings::HelpRequired)
+    ///     .arg(
+    ///         Arg::with_name("foo").help("It does foo stuff")
+    ///         // As required via AppSettings::HelpRequired, a help message was supplied
+    ///      )
+    /// # ;
+    /// ```
+    ///
+    /// # Panics
+    ///
+    /// ```rust,should_panic
+    /// # use clap::{App, Arg, AppSettings};
+    /// App::new("myapp")
+    ///     .setting(AppSettings::HelpRequired)
+    ///     .arg(
+    ///         Arg::with_name("foo")
+    ///         // Someone forgot to put .help("...") here
+    ///         // Since the setting AppSettings::HelpRequired is activated, this will lead to
+    ///         // a panic (if you are in debug mode)
+    ///     ).get_matches()
```

Comment:
You should hide the `.get_matches()` part.

Reply:
Hi @pksunkara ! Thanks for your feedback. 

True, I should get rid of this "get_matches()".

The "AppSetting" solution is taken from the issue #1639. You are right, that it will not be used during parsing. I am willing to do it differently but do not know how. What exactly do you mean by "feature on clap"?

Reply:
I will help you out regarding features. But first I would like opinion from @CreepySkeleton on what he thinks before we proceed forward. Thanks for the PR. 🎈

Reply:
Also cc @BurntSushi in case he wants to say something here regarding features.

Reply:
Thanks @pksunkara ! I just found some documentation regarding features in [https://doc.rust-lang.org/cargo/reference/manifest.html#the-features-section](url). Looks like a nice solution. If you and @CreepySkeleton decide it should be done via feature, I think I might be able to propose a solution without needing help.

Reply:
@thomasfermi It's explained [here](https://doc.rust-lang.org/cargo/reference/manifest.html#the-features-section) but I think your implementation is OK.